src/assignment/views/AssignmentEdit.tsx

```diff
@@ -634,6 +652,28 @@ const AssignmentEdit: FunctionComponent<DefaultSecureRouteProps<{ id: string }>>
 					</StickyEdgeBar>
 				)}
 			</Container>
+
+			{original && (
+				<ConfirmSlice
+					isOpen={!!isConfirmSliceOpen}
```

Comment:
You might not need a separate boolean to open the modal if you use:
```
isOpen={!!getConfirmSliceBlock}
```
Then you might be able to replace `useSingleEntityModal` with a simple useState:
```
const [confirmDeleteBlock, setConfirmDeleteBlock] = useState<AssignmentBlock | null>(null);

...

isOpen={!!confirmDeleteBlock}
```

Reply:
Dedicated boolean is in lijn met de rest van de applicatie, ik zie niet meteen plaatsen waar een null-check wordt gebruikt als trigger.